metrics/factory.go

```diff
@@ -27,11 +27,20 @@ type Options struct {
 	Help string
 }
 
+// HistogramOptions defines the information associated with a metric
+type HistogramOptions struct {
+	Name    string
+	Tags    map[string]string
+	Help    string
+	Buckets []float64
+}
+
 // Factory creates new metrics
 type Factory interface {
 	Counter(metric Options) Counter
 	Timer(metric Options) Timer
```

Comment:
my main question is: do we want to keep the Timer() as a function in the interface, or move it to a static helper function that will be based on the histogram? 

Pros: This is what already happens in most implementations, so we could just do it once.
Cons: in some metrics backends timers are 1st class citizens

Reply:
Your call :smile:

Reply:
so the only change I would propose is having the Timer() function take options struct that includes buckets (probably expressed as `[]time.Duration` rather than `[]float64`).